packages/data/addon/utils/adapter.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright 2019, Yahoo Holdings Inc.
+ * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
+ */
+import config from 'ember-get-config';
+import { warn } from '@ember/debug';
+import { getWithDefault } from '@ember/object';
+
+/**
+ * Gets host by name in [{name: String, uri: String}]
+ * config datastructure in config.navi.dataSources.
+ *
+ * @param {String} name - name of host to get
+ * @returns {String} - uri of fact datasource to use
+ */
+export function getHost(name) {
+  if (name) {
+    const host = config.navi.dataSources.find(dataSource => dataSource.name === name);
+    if (host && host.uri) {
+      return host.uri;
+    }
+    warn(`Fact host for ${name} requested but none was found in configuration. Falling back to default`, {
+      id: 'navi-fact-host-not-configured'
+    });
+  }
+  return config.navi.dataSources[0].uri;
+}
+
+/**
+ * Gets default data source from config
+ * @returns {String} - name of default data source
+ */
+export function getDefaultDataSource() {
+  return getWithDefault(config, 'navi.defaultDataSource', 'facts');
```

Comment:
Can you explain what the config should look like now? I don't believe we had `defaultDataSource` as a part of it previously.

Reply:
Yeah this is a new concept. old concept was just use the first dataSource provided, which defaulting by position when you plan on moving towards a named approach didn't work well.

right now, in the updated version of this patch.... it will fallback to the first if this config option isn't set, like the old behavior.  But you can, and probably should add a defaultDataSource to your config.js initializer.

edit: added new config option to `initializers/config.js` in app